### PR TITLE
Add ipython.vm

### LINKED ISSUE #286: Package idea: ipython
https://ipython.org/

likely just add to https://github.com/mandiant/VM-Packages/blob/main/packages/libraries.python3.vm/tools/modules.xml

### PR DESCRIPTION
**Pull request by d35ha**
Closes #286.

### CODE REVIEW

File: packages/ipython.vm/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+$ErrorActionPreference = 'Stop'
+Import-Module vm.common -Force -DisableNameChecking
+
+$toolName = 'ipython'
+$category = 'Productivity Tools'
```
Comment (Ana06) [inline packages/ipython.vm/tools/chocolateyinstall.ps1:5]: Do we need a shortcut for ipython in the Productivity Tools folder? I personally think we don't and that we should directly use `VM-Pip-Install ipython` instead. But what do others think? @mr-tz @mandiant/vms opinions?

**Review by Ana06 — COMMENTED** [review]
I have tested it locally and it works as expected :tada: not sure why it didn't work last time I tried it: https://github.com/mandiant/VM-Packages/issues/286#issuecomment-2233487344 :confused:

File: packages/ipython.vm/ipython.vm.nuspec [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>ipython.vm</id>
+    <version>8.27.0.20241001</version>
+    <authors>Fernando Perez</authors>
+    <description>IPython provides a rich toolkit to help you make the most out of using Python interactively</description>
```
[the remark this answers is not in the document]
Comment (d35ha) [reply packages/ipython.vm/ipython.vm.nuspec:7]: Done.

File: packages/ipython.vm/tools/chocolateyinstall.ps1 (hunk 1 above)
Comment (d35ha) [reply packages/ipython.vm/tools/chocolateyinstall.ps1:5]: @Ana06 
By default, ipython is installed and available from the command line (I believe it is a dependency for another installed python library). This package installs its shortcut under `Productivity Tools`.

Comment (Ana06) [reply packages/ipython.vm/tools/chocolateyinstall.ps1:5]: You should always test new packages in an empty VM as documented in https://github.com/mandiant/VM-Packages/wiki/Testing#local-testing. The users can select the packages to install when installing FLARE-VM and commando VM uses different packages as well. In a VM with FLARE-VM and only the following packages I do not have ipython available:
```
Boxstarter|3.0.3
Boxstarter.Bootstrapper|3.0.3
Boxstarter.Chocolatey|3.0.3
Boxstarter.Common|3.0.3
Boxstarter.HyperV|3.0.3
Boxstarter.WinConfig|3.0.3
chocolatey|2.3.0
chocolatey-compatibility.extension|1.0.0
chocolatey-core.extension|1.4.0
chocolatey-windowsupdate.extension|1.0.5
common.vm|0.0.0.20240913
debloat.vm|0.0.0.20240327
hashmyfiles.vm|0.0.0.20240411
installer.vm|0.0.0.20240402
KB2919355|1.0.20160915
KB2919442|1.0.20160915
KB2999226|1.0.20181019
KB3033929|1.0.5
KB3035131|1.0.3
pma-labs.vm|0.0.0.20240411
python3|3.10.11
python3.vm|0.0.0.20240726
python310|3.10.11
vcredist140|14.40.33810
vcredist2015|14.0.24215.20170201
```

I hadn't noticed that the default configuration installs ipython. I wonder which package is installing it. Any clues @mandiant/vms?

Comment (emtuls) [reply packages/ipython.vm/tools/chocolateyinstall.ps1:5]: @d35ha Is the shortcut useful? 

I personally haven't used iPython directly, I think only as part of Jupyter Notebooks, so I'm not too familiar with how often it gets used directly.

Comment (d35ha) [reply packages/ipython.vm/tools/chocolateyinstall.ps1:5]: I normally use it, it is a way better than the traditional python shell.

**Review by emtuls — APPROVED** [review]

File: packages/ipython.vm/tools/chocolateyinstall.ps1 (hunk 1 above)
Comment (Ana06) [reply packages/ipython.vm/tools/chocolateyinstall.ps1:5]: @d35ha I assume you mean you use IPython and not the shortcut. I personally think IPython does not need a shortcut and that this package should only install it.

**Pull request merged**